pallets/did/src/did_details.rs

```diff
@@ -559,28 +527,13 @@ pub struct DidCreationDetails<T: Config> {
 	pub did: DidIdentifierOf<T>,
 	/// The authorised submitter of the creation operation.
 	pub submitter: AccountIdOf<T>,
-	/// The new key agreement keys.
-	pub new_key_agreement_keys: DidNewKeyAgreementKeySet<T>,
-	/// \[OPTIONAL\] The new attestation key.
-	pub new_attestation_key: Option<DidVerificationKey>,
-	/// \[OPTIONAL\] The new delegation key.
-	pub new_delegation_key: Option<DidVerificationKey>,
-	/// The service endpoints details.
-	pub new_service_details: Vec<DidEndpoint<T>>,
```

Comment:
Good point. The question is whether we want to bump it in each PR on develop, or whether we should bump it once before releasing, as we do with the spec version and everything else. I would not do something different just for the tx version. Should we add it to the checklist?

Reply:
IMO we should bump it in each PR to develop.

Reply:
But why don't we do the same with the spec version then? How can one know whether the last time the version was bumped was before or after the release?

Reply:
For me, the difference is that bumping the spec version is mandatory before each runtime upgrade. Bumping the `transaction_version` however happens rather rarely. If we postpone bumping this, we might be missing it in the release. I don't like relying on adding it to the release checklist. Consider a longer timespan between merging such a PR and doing a release and a release captain who has little to no clue about this PR.

IMO, bumping in the corresponding PR is more efficient and less error-prone.

Reply:
Both options are not perfect. I'm currently preferring not increasing the transaction_version in the PR, but cloning the checklist template and already prepare it for the next release by adding a check for the transaction_version.

In the long run we need a release pipeline that checks the Metadata for breaking changes.

Reply:
I think we have two options here.
One would be to make better use of labels, so that we can mark PRs that have breaking changes and verify at the beginning of the release cycle whether we need to bump, or we check whenever we merge from `develop` into `master` whether the tx version has been bumped at least in one PR (meaning we would have to bump it each time), and "adjust" it to the next value.

Reply:
We definitely need to improve our release pipeline. I'm also in favor of improving our labels and adding more.

Reply:
For the time being, I am bumping the tx_version https://github.com/KILTprotocol/mashnet-node/pull/350/commits/4f144aa58f97c22038dda6ae03e50e5a8892f226. Until we have a good automation in place, I would then suggest to bump them in each PR that needs it, and then adjust the final value at the beginning of each release cycle, if needed.